src/VerifyAccessToken.php

```diff
@@ -86,50 +67,65 @@ protected function getNewAccessToken(): string
      */
     public function handle($request, Closure $next, ...$scopes)
     {
-        $authorization = $request->header('Authorization');
-
-        if (!$authorization) {
-            throw new InvalidInputException('No Authorization header present');
-        }
-
         $bearerToken = $request->bearerToken();
 
         if (!$bearerToken) {
-            throw new InvalidInputException('No Bearer token in the Authorization header present');
+            throw new AuthenticationException('No Bearer token present');
         }
 
         try {
             $result = $this->getIntrospect($bearerToken);
 
             if (!$result['active']) {
-                throw new InvalidAccessTokenException('Invalid token!');
+                throw new AuthenticationException('Invalid token!');
             }
 
-            if ($scopes != null) {
-                if (!\is_array($scopes)) {
-                    $scopes = [$scopes];
-                }
-
-                $scopesForToken = \explode(' ', $result['scope']);
-
-                if (count($misingScopes = array_diff($scopes, $scopesForToken)) > 0) {
-                    throw new InvalidAccessTokenException('Missing the following required scopes: ' . implode(' ,', $misingScopes));
-                }
+            if ($scopes !== null) {
+                $this->checkScopes(explode(' ', $result['scope']), $scopes);
             }
         } catch (RequestException $e) {
             if ($e->hasResponse()) {
                 $result = json_decode((string) $e->getResponse()->getBody(), true);
 
                 if (isset($result['error'])) {
-                    throw new InvalidAccessTokenException($result['error']['title'] ?? 'Invalid token!');
+                    throw new AuthenticationException($result['error']['title']);
```

Comment:
do you want the `?? 'Invalid token!`?

Reply:
Oh, I think I do actually, I mis-read it